Unable to scroll results.webkit.org results using the scrollbars

smfr opened the pull request:
#### db2f6b2a047a9f2f493bd20a6f0faff640fd3f8e
<pre>
Unable to scroll results.webkit.org results using the scrollbars
<a href="https://bugs.webkit.org/show_bug.cgi?id=237308">https://bugs.webkit.org/show_bug.cgi?id=237308</a>
rdar://89598421

Reviewed by Dan Glastonbury and Richard Robinson.

If an overflow:scroll contained a composited, positioned descendant, then trying to click and drag an
overlay scrollbar that overlapped that descendant would fail to work. This affected results.webkit.org,
where the canvas elements caused the bug.

This happened because when the hit-testing code tested whether the composited layer contained the
point, it would use a backgroundClipRect that was not shrunk for the scrollbar. There was already
a &quot;FIXME&quot; comment on a line of code that explicitly removed the `IncludeOverlayScrollbarSize` option,
and removing this line fixes the bug. The code dates from when composited overflow scrolling was
first implemented (112856@main). `IncludeOverlayScrollbarSize` is only ever specified when requesting
clip rects for hit-testing, so it seems reasonable to assume the code was wrong.

Do other whitespace cleanup in the hit-testing code, including fixing a line with bad indentation.

* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-hit-test-expected.txt: Added.
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-hit-test.html: Added.
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::hitTest):
(WebCore::RenderLayer::hitTestLayer):
(WebCore::RenderLayer::hitTestContentsForFragments const):
(WebCore::RenderLayer::calculateClipRects const):

Canonical link: <a href="https://commits.webkit.org/269255@main">https://commits.webkit.org/269255@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b94f9a52809b7a103cfe1352f2c952398fd0efe1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21990 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/22205 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23057 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23872 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20360 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22234 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26452 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22518 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21433 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22219 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21862 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19068 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24724 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18979 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19922 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26186 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20003 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20146 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24052 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20583 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17532 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19936 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/19731 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5251 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/24141 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20534 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-hit-test.html:
```diff
@@ -0,0 +1,97 @@
+<html>
+<head>
+    <style>
+        .scroller {
+	        overflow-x: auto;
+	        overflow-y: hidden;
+	        width: 500px;
+	        height: 200px;
+	        border: 1px solid black;
+			will-change: z-index;
+			resize: both;
+        }
+	    .wide {
+	        position: relative;
+	        width: 400%;
+	        border: 1px solid green;
+	    }
+        .sticky {
+	        position: sticky;
+	        top: 10px;
+	        left: 10px;
+	        width: 350px;
+	        height: 200px;
+	        background-color: rgba(0, 0, 0, 0.2);
+        }
```
Nit: Indentation is inconsistent.

smfr committed:
Unable to scroll results.webkit.org results using the scrollbars
https://bugs.webkit.org/show_bug.cgi?id=237308
rdar://89598421

Reviewed by Dan Glastonbury and Richard Robinson.

If an overflow:scroll contained a composited, positioned descendant, then trying to click and drag an
overlay scrollbar that overlapped that descendant would fail to work. This affected results.webkit.org,
where the canvas elements caused the bug.

This happened because when the hit-testing code tested whether the composited layer contained the
point, it would use a backgroundClipRect that was not shrunk for the scrollbar. There was already
a "FIXME" comment on a line of code that explicitly removed the `IncludeOverlayScrollbarSize` option,
and removing this line fixes the bug. The code dates from when composited overflow scrolling was
first implemented (112856@main). `IncludeOverlayScrollbarSize` is only ever specified when requesting
clip rects for hit-testing, so it seems reasonable to assume the code was wrong.

Do other whitespace cleanup in the hit-testing code, including fixing a line with bad indentation.

* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-hit-test-expected.txt: Added.
* LayoutTests/fast/scrolling/mac/scrollbars/overflow-overlay-scrollbar-hit-test.html: Added.
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::hitTest):
(WebCore::RenderLayer::hitTestLayer):
(WebCore::RenderLayer::hitTestContentsForFragments const):
(WebCore::RenderLayer::calculateClipRects const):

Canonical link: https://commits.webkit.org/269255@main
